packages/framework/esm-styleguide/src/location-selector/location-selector.component.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ComboBox } from '@carbon/react';
+import { useLocationByUuid, useLocations } from '../location-picker/location-picker.resource';
+import styles from './location-selector.module.scss';
+
+interface LocationSelectorProps {
+  selectedLocationUuid?: string;
+  defaultLocationUuid?: string;
+  locationTag?: string;
+  locationsPerRequest?: number;
+  onChange: (locationUuid?: string) => void;
+  Locationlabel: string;
+  comBoxLabel: string;
+}
+
+export const LocationSelector: React.FC<LocationSelectorProps> = ({
+  selectedLocationUuid,
+  defaultLocationUuid,
+  locationTag,
+  locationsPerRequest = 50,
+  onChange,
+  Locationlabel,
+  comBoxLabel,
+}) => {
+  const { t } = useTranslation();
+  const [searchTerm, setSearchTerm] = useState<string>('');
+
+  let defaultLocation = useLocationByUuid(defaultLocationUuid).location;
+
+  const { locations: fetchedLocations } = useLocations(locationTag, locationsPerRequest, searchTerm);
+
+  const locations = useMemo(() => {
+    if (defaultLocation && !searchTerm) {
+      return [defaultLocation, ...fetchedLocations?.filter(({ resource }) => resource.id !== defaultLocationUuid)];
+    }
+    return fetchedLocations;
+  }, [defaultLocation, fetchedLocations]);
+
+  const handleSearch = (searchString) => {
+    //debounce the search input and call api
```

Comment:
I guess this is a todo comment. If so, do this:
```suggestion
    // TODO: <ticket number for the todo> debounce the search input and call api
```

Reply:
Ah never mind, no need a ticket id since this is still a WIP.